### ISSUE TITLE
Disable link checking for GitHub Alerts

### ISSUE THREAD

**Issue by gabeseltzer** [DavidAnson/vscode-markdownlint#359]
GitHub uses the following syntax for ["Alerts"](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts):

```markdown
> [!NOTE]
> Useful information that users should know, even when skimming content.
```

markdownlint will flag the `[!NOTE]` as a link, and will tell you `No link definition found: '!NOTE'(link.no-such-reference)`. Can we turn this off for the keywords that make up alerts? Specifically: 
* `[!NOTE]`
* `[!TIP]`
* `[!IMPORTANT]`
* `[!WARNING]`
* `[!CAUTION]`
